quickwit-cli/src/index.rs

```diff
@@ -0,0 +1,1004 @@
+    let split_meta_and_footers = metastore
+        .list_splits(&args.index_id, SplitState::Published, None, &[])
+        .await?;
+    let demux_uniq_values: HashSet<String> = split_meta_and_footers
+        .iter()
+        .map(|metadata_and_footer| {
+            metadata_and_footer
+                .split_metadata
+                .tags
+                .iter()
+                .filter(|tag| match_tag_field_name(&demux_field_name, tag))
+                .cloned()
+        })
+        .flatten()
+        .collect();
+    println!("{}", "Statistic reports on demux:".bold());
+    println!(
+        "{}",
+        format!(
+            "- Found {} `{}` unique values in {} splits.",
+            demux_uniq_values.len(),
+            demux_field_name,
+            split_meta_and_footers.len(),
+        )
+    );
+    // Compute split count per demux value.
+    println!(
+        "{}",
+        format!("- Stats on split count per `{}` value.", demux_field_name)
+    );
+    let mut split_counts_per_demux_values = Vec::new();
+    for demux_value in demux_uniq_values {
+        let split_count = split_meta_and_footers
+            .iter()
+            .filter(|split_meta_and_footer| {
+                split_meta_and_footer
+                    .split_metadata
+                    .tags
+                    .contains(&demux_value)
+            })
+            .count();
+        split_counts_per_demux_values.push(split_count);
+    }
+    print_demux_stats(&split_counts_per_demux_values);
+
+    // Compute demux unique values count per split.
+    let (non_demuxed_splits, demuxed_splits): (Vec<_>, Vec<_>) = split_meta_and_footers
+        .iter()
+        .cloned()
+        .map(|metadata_and_footer| metadata_and_footer.split_metadata)
+        .partition(|split| split.demux_num_ops == 0);
+    let non_demuxed_split_demux_values_counts = non_demuxed_splits
+        .iter()
+        .map(|split| {
+            split
+                .tags
+                .iter()
+                .filter(|tag| match_tag_field_name(&demux_field_name, tag))
+                .count()
+        })
+        .sorted()
+        .collect_vec();
+    let demuxed_split_demux_values_counts = demuxed_splits
+        .iter()
+        .map(|split| {
+            split
+                .tags
+                .iter()
+                .filter(|tag| match_tag_field_name(&demux_field_name, tag))
+                .count()
+        })
+        .sorted()
+        .collect_vec();
+    println!(
+        "{}",
+        format!(
+            "- Stats on `{}` unique values count per split.",
+            demux_field_name
+        )
+    );
+    println!("  -> On {} non demuxed splits:", non_demuxed_splits.len());
+    print_demux_stats(&non_demuxed_split_demux_values_counts);
+    println!("  -> On {} demuxed splits:", demuxed_splits.len());
+    print_demux_stats(&demuxed_split_demux_values_counts);
+
+    Ok(())
+}
+
+fn print_demux_stats(counts: &[usize]) {
```

Comment:
I would suggest moving all demux stats calculation helper functions to another file.